File: June Long Challenge 2019/RSIGNS.py
# for _ in range(int(input())):
#     n = int(input())

#     print(5*pow(2, n))

MOD = 1000000007
arr = {}

arr[1] = 10

for _ in range(int(input())):
    n = int(input())

    # for i in range(1, n):
    #     print(5*pow(2, i))

    if arr.get(n):
        print(arr[n]) if(n==1) else print(arr[n]-1)
    else:
        for i in range(list(arr.keys())[-1]+1, n+1):
            arr[i] = (arr[i-1] + 10*pow(2, i-2))%MOD
        print(arr[n]) if(n==1) else print(arr[n]-1)

File: June Long Challenge 2019/RSIGNS.cpp
#include<bits/stdc++.h>
using namespace std;

#define lli long long int

int main()
{
    lli MOD = 1000000007;
    lli arr[1000000];
    arr[1] = 10;

    lli test; cin >> test;
    while(test--)
    {
        lli k; cin >> k;

        if(arr[k])
            cout << arr[k]%MOD << endl;
        else
        {
            for(lli i=2; i<k+1; i++)
                arr[i] = (arr[i-1] + 10*(lli)(pow(2, i-2)))%MOD;
            cout << arr[k]%MOD << endl;
        }
    }
}